lib/game_client/ogs/ogs_game.dart

```diff
@@ -575,6 +587,133 @@ class OGSGame extends Game {
     );
   }
 
+  /// Convert current game state to 2D array format expected by AI server
+  /// 0 = empty, 1 = black, 2 = white
+  List<List<int>> _getCurrentBoardStateForAIServer() {
+    final boardState = BoardState(size: boardSize);
+
+    // Replay all moves to get current board state
+    for (final move in _allMoves) {
+      boardState.move(move);
+    }
+
+    final board = generate2D<int>(
+        boardSize,
+        (row, col) => switch (boardState[(row, col)]) {
+              null => 0,
+              wq.Color.black => 1,
+              wq.Color.white => 2,
+            });
+
+    return board;
+  }
+
+  /// Gives the rules string expected by OGS APIs
+  String _rulesToOgsString(Rules rules) => switch (rules) {
+        Rules.chinese => 'chinese',
+        Rules.japanese => 'japanese',
+        Rules.korean => 'korean',
+      };
+
+  /// Gets a score evaluation from the server
+  Future<Map<String, dynamic>?> _callAIServer(
+      List<List<int>> boardState, String playerToMove) async {
+    _logger.fine('Calling AI server for game $id');
+
+    final url = Uri.parse('$_aiServerUrl/api/score');
+
+    final payload = {
+      'player_to_move': playerToMove,
+      'width': boardSize,
+      'height': boardSize,
+      'rules': _rulesToOgsString(rules),
+      'board_state': boardState,
+      'autoscore': true,
+      'jwt': _jwtToken,
+    };
+
+    try {
+      final response = await http.post(
+        url,
+        headers: {
+          'Content-Type': 'application/json',
+          'User-Agent': 'WeiqiHub/1.0',
```

Comment:
should we reuse the constant from `OGSGameClient`? https://github.com/ale64bit/WeiqiHub/blob/main/lib/game_client/ogs/ogs_game_client.dart#L29. It can be made public, since it's a `const` anyway.